tests/csapi/direct_message_create_test.go

```diff
@@ -0,0 +1,56 @@
+package csapi_tests
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/tidwall/gjson"
+)
+
+func TestDirectMessageCreate(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintOneToOneRoom)
+	defer deployment.Destroy(t)
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	bob := deployment.Client(t, "hs1", "@bob:hs1")
+
+	// Alice invites Bob to a room for direct messaging.
+	roomID := alice.CreateRoom(t, map[string]interface{}{
+		"invite":    []string{bob.UserID},
+		"is_direct": true,
+	})
+
+	// Bob receives the invite and joins.
+	next_batch := bob.SyncUntilInvitedTo(t, roomID)
+	bob.JoinRoom(t, roomID, []string{"hs1"})
+
+	// Wait until Bob sees that he's joined.
+	isBobJoinEvent := func(ev gjson.Result) bool {
+		return ev.Get("type").Str == "m.room.member" && ev.Get("state_key").Str == bob.UserID && ev.Get("content").Get("membership").Str == "join"
+	}
+	bob.SyncUntilTimelineHas(t, roomID, isBobJoinEvent)
+
+	// Repeat the sync, but in a form where we can inspect the output.
```

Comment:
This doesn't make sense because `SyncUntilTimelineHas` returns each event in the callback `func(ev gjson.Result)`. You can do any inspection or assertion you like on the event, and indeed you already do on `:30` to check the type/state_key/membership.